prow/plugins/lifecycle/lifecycle.go

```diff
@@ -121,10 +125,34 @@ func handleOne(gc lifecycleClient, log *logrus.Entry, e *github.GenericCommentEv
 	remove := mat[1] != ""
 	cmd := mat[2]
 	lbl := "lifecycle/" + cmd
-	// Let's start simple and allow anyone to add/remove frozen, stale, putrid, rotten labels.
+
+	// Let's start simple and allow anyone to add/remove frozen, stale, rotten labels.
 	// Adjust if we find evidence of the community abusing these labels.
-	if remove {
+	labels, err := gc.GetIssueLabels(org, repo, number)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to get labels.")
+	}
+
+	// If the label exists and we asked for it to be removed, remove it.
+	if github.HasLabel(lbl, labels) && remove {
 		return gc.RemoveLabel(org, repo, number, lbl)
 	}
-	return gc.AddLabel(org, repo, number, lbl)
+
+	// If the label does not exist and we asked for it to be added, add it.
+	if !github.HasLabel(lbl, labels) && !remove {
+		if err := gc.AddLabel(org, repo, number, lbl); err != nil {
```

Comment:
Would we want to remove the old labels before adding the new one?

Reply:
> Would we want to remove the old labels before adding the new one?

Done. Removes old labels first, then adds the new one.